api_app/analyzers_manager/observable_analyzers/dns/dns_malicious_detectors/spamhaus_wqs.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+from ..dns_responses import malicious_detector_response
+
+logger = logging.getLogger(__name__)
+
+
+class SpamhausWQS(classes.ObservableAnalyzer):
+    url: str = "https://apibl.spamhaus.net/lookup/v1"
+    _api_key: str = None
+
+    def update(self):
+        pass
+
+    def run(self):
+        headers = {"Authorization": f"Bearer {self._api_key}"}
+
+        if self.observable_classification == self.ObservableTypes.DOMAIN.value:
+            response = requests.get(
+                url=f"{self.url}/DBL/{self.observable_name}", headers=headers
+            )
+        elif self.observable_classification == self.ObservableTypes.IP.value:
+            response = requests.get(
+                url=f"{self.url}/AUTHBL/{self.observable_name}", headers=headers
+            )
```

Comment:
you could try to simplify this by using the if/else inside the { } and changing the string there.